cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -444,7 +444,11 @@ public static KafkaCluster fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup vers
         } else {
             metricReporterList.remove(CRUISE_CONTROL_METRIC_REPORTER);
         }
-        configuration.setConfigOption(KAFKA_METRIC_REPORTERS_CONFIG_FIELD, String.join(",", metricReporterList));
+        if (!metricReporterList.isEmpty()) {
+            configuration.setConfigOption(KAFKA_METRIC_REPORTERS_CONFIG_FIELD, String.join(",", metricReporterList));
+        } else {
+            configuration.removeConfigOption(KAFKA_METRIC_REPORTERS_CONFIG_FIELD);
+        }
```

Comment:
out of curiosity, how this is related to CC?

Reply:
When creating the Cruise Control deployment, we want to update the Kafka `metric.reporter` configuration without overwriting any existing metric reporters that are already listed, the first part of the conditional appends the cruise control metric reporter to the list of `metric.reporters`. 

When deleting the Cruise Control deployment, we want to update the Kafka `metric.reporter` configuration to remove the cruise control metric reporter. The second part of the conditional ensures that an empty `metric.reporters=` field is removed from Kafka configuration when we remove the cruise control metric reporter.

Reply:
sorry I forgot the `metric.reporter` :-)